codegen/src/lib.rs

```diff
@@ -8,6 +8,20 @@ use quote::quote;
 use syn::Error;
 use syn::ItemFn;
 
+//We can't use procedural macros as expressions due to rust limitations
+//so this generates a function instead
+#[proc_macro]
+pub fn make_env(input: TokenStream) -> TokenStream {
```

Comment:
A real Rustdoc and a speaking method name would be great.

Reply:
Can you clarify what you mean by speaking method name?

Reply:
Maybe `make_read_env_var`?